pkg/connector/team.go

```diff
@@ -266,13 +267,20 @@ func (o *teamResourceType) Grant(ctx context.Context, principal *v2.Resource, en
 		return nil, fmt.Errorf("github-connectorv2: failed to get user %d, err: %w", userId, err)
 	}
 
+	enIDParts := strings.Split(entitlement.Id, ":")
```

Comment:
LGTM. but, I noticed we have the same code twice, it would be a good idea to reuse this code and add a helper or a function.  :)